Post: TIL Cleaner queries with Ecto map

yuriploc opened the pull request:
Hello there!

Sharing a TIL based on a discussion we had during a mentorship meeting.

Please advise of any needed changes.

yuriploc commented on posts/2021-11-18-til-cleaner-queries-with-ecto-map.md:
```diff
@@ -0,0 +1,75 @@
+
+---
+
+# TIL: Cleaner queries with Ecto `map`
```
cc @SophieDeBenedetto

SophieDeBenedetto commented on posts/2021-11-18-til-cleaner-queries-with-ecto-map.md:
```diff
@@ -0,0 +1,75 @@
+date: ~D[2021-11-18],
+title: "TIL: Cleaner queries with Ecto `map`",
+excerpt: """
+How learning more about the Ecto.Query module can help you write cleaner select queries.
```
I might sum this up in a way that is a bit more specific to this post's content. Maybe something like: "Learn how you can write cleaner Ecto select queries with the help `Ecto.Query.map`"

SophieDeBenedetto commented on posts/2021-11-18-til-cleaner-queries-with-ecto-map.md:
````diff
@@ -0,0 +1,75 @@
+Repo.all(query)
+```
+
+And getting back more data than we would care using, we can explicitly tell Ecto (and the DB) which columns we want it to return us:
````
Reword as: "To avoid getting back more data than we need,..."

SophieDeBenedetto commented on posts/2021-11-18-til-cleaner-queries-with-ecto-map.md:
```diff
@@ -0,0 +1,75 @@
+
+# TIL: Cleaner queries with Ecto `map`
+
+It's not unusual for database queries to avoid select * when we can use an index (say, :id) and have a performance gain.
```
I don't think its necessarily more performant from a DB query execution perspective to be explicit vs. using `SELECT *`. But, there are definite readability and maintenance gains to being explicit. And, in the case of loading many columns into memory, there may be an application performance gain to being specific. I would re-word this a bit since saying "...when we can use an index" makes it seem like its a DB perf gain related to leveraging DB indexes.

yuriploc replied:
The sentence may not read well, but the idea is by replacing `SELECT *` for a `SELECT id, ...` we can leverage optimal indexes and have a performance gain, summing with the advantages you also mentioned. I'll try to work more on this, thank you.

SophieDeBenedetto commented on posts/2021-11-18-til-cleaner-queries-with-ecto-map.md:
```diff
@@ -0,0 +1,75 @@
+
+But why do we have to be so explicit and duplicate keys and values? Isn't there a better way?
+
+It turns out Ecto.Query already solved this for us with the map/2 function. So this:
```
"map/2" -> `map/2` (put the function name in backticks)

SophieDeBenedetto commented on posts/2021-11-18-til-cleaner-queries-with-ecto-map.md:
````diff
@@ -0,0 +1,75 @@
+    |> Repo.all()
+end
+```
+
````
This is SUCH a nice clean piece of code. I'd add one sentence to wrap up/conclude here. Something like: "Thanks to the `Ecto.Query.map/2` function and the use of pipes, we end up with clean, composable and highly readable code".

SophieDeBenedetto reviewed (changes requested):
This looks great and ✨ 🎉 on your first contribution @yuriploc! I left a few comments, please let me know if you have any questions 😄

SophieDeBenedetto reviewed (approved):
LGTM! I'll cc @elixirschool/english to see if we can get one more set of 👀  and then we can launch it 🚀 😄

doomspork reviewed (approved):
This looks great @yuriploc! What is your twitter handle?

doomspork commented:
We should rename the file and update the date to reflect our merge date, but I can do that quickly post-merge 👍

yuriploc commented:
> We should rename the file and update the date to reflect our merge date, but I can do that quickly post-merge +1

`yuriploc` is my twitter handle as well. Just updated the date and renamed the file, thank you.

doomspork commented:
Thank you again @yuriploc! https://twitter.com/elixirschool/status/1462847051860004864
